lib/matplotlib/ticker.py

```diff
@@ -1992,19 +1992,16 @@ def __init__(self, nbins=None, **kwargs):
         symmetric : bool, default: False
             If True, autoscaling will result in a range symmetric about zero.
 
-        prune : {'lower', 'upper', 'both', None}, default: None
-            Remove edge ticks -- useful for stacked or ganged plots where
-            the upper tick of one axes overlaps with the lower tick of the
-            axes above it, primarily when :rc:`axes.autolimit_mode` is
-            ``'round_numbers'``.  If ``prune=='lower'``, the smallest tick will
-            be removed.  If ``prune == 'upper'``, the largest tick will be
-            removed.  If ``prune == 'both'``, the largest and smallest ticks
-            will be removed.  If *prune* is *None*, no ticks will be removed.
-
         min_n_ticks : int, default: 2
             Relax *nbins* and *integer* constraints if necessary to obtain
             this minimum number of ticks.
         """
+        if 'prune' in kwargs:
```

Comment:
perhaps push this warning into `set_params`, as it should probably be deprecated for setting after init as well (and init just calls set_params, anyway)

Reply:
👍 good idea